tools/SourceKit/lib/SwiftLang/SwiftASTManager.cpp

```diff
@@ -83,7 +83,7 @@ struct InvocationOptions {
     // Assert invocation with a primary file. We want to avoid full typechecking
     // for all files.
     assert(!this->PrimaryFile.empty());
-    assert(this->Invok.getFrontendOptions().Inputs.hasPrimaryInput());
+    assert(this->Invok.getFrontendOptions().Inputs.havePrimaryInputs());
```

Comment:
This probably wants the stronger assertion of *exactly one* primary file. (This is the entry point used for code completion, among other things.)

Reply:
Fixed. Encapsulated predicate in haveUniquePrimaryInput, used haveUniquePrimaryInput everywhere it applies, and changed all has... to have... routines in Inputs (to be consistent).